actiontext/app/javascript/actiontext/attachment_upload.js

```diff
@@ -20,13 +20,14 @@ export class AttachmentUpload {
 
   directUploadDidComplete(error, attributes) {
     if (error) {
-      throw new Error(`Direct upload failed: ${error}`)
+      let uploadFailed = new CustomEvent("trix-attachment-error", {"detail": {"error": error, "attachment": this.attachment }})
```

Comment:
The name `trix-attachment-error` suggests the event comes from Trix itself (all of Trix's events start with `trix-*`). I'd prefer we follow Active Storage's convention and name it `direct-upload:error`: https://github.com/rails/rails/blob/3d0a7822a5dc8828a32c9bf032daa950223364a2/activestorage/app/javascript/activestorage/direct_upload_controller.js#L51 https://github.com/rails/rails/blob/3d0a7822a5dc8828a32c9bf032daa950223364a2/activestorage/app/javascript/activestorage/direct_upload_controller.js#L47

Reply:
Hi Javan! Thanks for taking the time to take a look at this, I really appreciate it. 

I must admit I thought something similar, using the trix namespace didn't feel right in this scope, however in terms of usage it is usually paired with trix-attachment-add/remove. I think looking at it again, if you're happy with it then it does make more sense from within this library. The tradeoff is in the typical usage, where you'll be dealing with two different event 'namespaces' - the trix-* and direct-upload-*.  

My initial thought around using ActiveStorage was that I'd need to alter that to export the dispatchEvent method, let's give it a try!